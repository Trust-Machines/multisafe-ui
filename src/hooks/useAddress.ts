import useUserSession from './useUserSession';
import useNetwork from './useNetwork';

const useAddress = (): string | null => {
    const [, userData] = useUserSession();
    const [network] = useNetwork();

    if (userData) {
        return userData.profile.stxAddress[network];
    }

    return null;
}

export default useAddress;