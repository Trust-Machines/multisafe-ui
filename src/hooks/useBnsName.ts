import {useState, useEffect} from 'react';

import useAddress from './useAddress';
import useNetwork from './useNetwork';
import {getBnsName} from '../api';

const useBnsName = (): string | null => {
    const [name, setName] = useState<string | null>(null);
    const address = useAddress();
    const [network, stacksNetwork] = useNetwork()

    useEffect(() => {
        setName(null);
        if (!address) {
            return;
        }

        getBnsName(stacksNetwork, address).then((r)=>{
            if(r){
                setName(r);
            }
        })
    }, [address, network, stacksNetwork])

    return name;
}

export default useBnsName;