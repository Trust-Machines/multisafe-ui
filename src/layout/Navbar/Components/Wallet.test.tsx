import {render} from '@testing-library/react';

import Wallet, {WalletMenu} from './Wallet';
import useAddress from '../../../hooks/useAddress';
import useBnsName from '../../../hooks/useBnsName';

jest.mock('../../../hooks/useAddress');
jest.mock('../../../hooks/useBnsName');

test('1 Default render', () => {
    const rendered = render(<Wallet/>);
    expect(rendered.container).toMatchSnapshot()
});

test('2 Render with address', () => {
    (useAddress as jest.Mock).mockReturnValue('SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7')
    const rendered = render(<Wallet/>);
    expect(rendered.container).toMatchSnapshot()
});

test('3 Render with bns', () => {
    (useAddress as jest.Mock).mockReturnValue('SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7');
    (useBnsName as jest.Mock).mockReturnValue('tbb.btc')
    const rendered = render(<Wallet/>);
    expect(rendered.container).toMatchSnapshot()
});

test('4 Default render WalletMenu', () => {
    const rendered = render(<WalletMenu/>);
    expect(rendered.container).toMatchSnapshot()
});


