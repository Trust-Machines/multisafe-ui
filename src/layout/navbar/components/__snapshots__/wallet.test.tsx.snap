// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 Default render 1`] = `
<div>
  <div
    class="MuiBox-root css-7lmw8c"
  >
    <div
      class="MuiBox-root css-198xhax"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18wkfow-MuiSvgIcon-root"
        data-testid="AccountBalanceWalletIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 18v1c0 1.1-.9 2-2 2H5c-1.11 0-2-.9-2-2V5c0-1.1.89-2 2-2h14c1.1 0 2 .9 2 2v1h-9c-1.11 0-2 .9-2 2v8c0 1.1.89 2 2 2h9zm-9-2h10V8H12v8zm4-2.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
        />
      </svg>
      <div
        class="MuiBox-root css-x1k79c"
      >
        <span
          class="first-line"
        >
          Not connected
        </span>
        <span
          class="second-line"
        >
          Connect Wallet
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`2 Render with address 1`] = `
<div>
  <div
    class="MuiBox-root css-7lmw8c"
  >
    <div
      class="MuiBox-root css-198xhax"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18wkfow-MuiSvgIcon-root"
        data-testid="AccountBalanceWalletIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 18v1c0 1.1-.9 2-2 2H5c-1.11 0-2-.9-2-2V5c0-1.1.89-2 2-2h14c1.1 0 2 .9 2 2v1h-9c-1.11 0-2 .9-2 2v8c0 1.1.89 2 2 2h9zm-9-2h10V8H12v8zm4-2.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
        />
      </svg>
      <div
        class="MuiBox-root css-x1k79c"
      >
        <span
          class="first-line"
        >
          SP3XD...KAEK7
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`3 Render with bns 1`] = `
<div>
  <div
    class="MuiBox-root css-7lmw8c"
  >
    <div
      class="MuiBox-root css-198xhax"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18wkfow-MuiSvgIcon-root"
        data-testid="AccountBalanceWalletIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 18v1c0 1.1-.9 2-2 2H5c-1.11 0-2-.9-2-2V5c0-1.1.89-2 2-2h14c1.1 0 2 .9 2 2v1h-9c-1.11 0-2 .9-2 2v8c0 1.1.89 2 2 2h9zm-9-2h10V8H12v8zm4-2.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
        />
      </svg>
      <div
        class="MuiBox-root css-x1k79c"
      >
        <span
          class="first-line"
        >
          SP3XD...KAEK7
        </span>
        <span
          class="second-line"
        >
          tbb.btc
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`4 Default render WalletMenu 1`] = `
<div>
  <div
    class="MuiBox-root css-9x36y4"
  >
    <div
      class="MuiBox-root css-eg13y7"
    >
      <span
        aria-label="Copied"
        class=""
        data-mui-internal-clone-element="true"
        style="cursor: pointer;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-5oa1ix-MuiSvgIcon-root"
          data-testid="ContentCopyIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
          />
        </svg>
        Copy Address
      </span>
    </div>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Logout
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
